disperser/batcher/txn_manager.go

```diff
@@ -186,19 +192,24 @@ func (t *txnManager) monitorTransaction(ctx context.Context, req *TxnRequest) (*
 			err = t.ethClient.SendTransaction(ctx, newTx)
 			if err != nil {
 				if errors.Is(err, gcore.ErrNonceTooLow) {
-					// try to get the receipt again to see if the transaction has been mined
-					_, receiptErr := t.ethClient.TransactionReceipt(ctx, req.Tx.Hash())
-					if receiptErr == nil {
-						continue
+					// try to get the receipt again to see if any of the transaction attempts have been mined
+					for _, tx := range req.txAttempts {
+						_, receiptErr := t.ethClient.TransactionReceipt(ctx, tx.Hash())
+						if receiptErr == nil {
+							// tx has been mined, update the request with the mined tx and continue
+							req.Tx = tx
+							continue
```

Comment:
Should it just break here since there is already a mined txn found, and the loop again to wait for more confirmations for that txn?

Reply:
Yes! Thanks for catching that